icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -0,0 +1,740 @@
+            this.fSource = identifier;
+
+            if (trie != null) {
+                return;
+            }
+
+
+
+            // Building the trie.
+            CharsTrieBuilder trieBuilder;
+            trieBuilder = new CharsTrieBuilder();
+
+
+            // Add syntax parts (compound, power prefixes)
+            trieBuilder.add("-per-", CompoundPart.COMPOUND_PART_PER.getTrieIndex());
+            trieBuilder.add("-", CompoundPart.COMPOUND_PART_TIMES.getTrieIndex());
+            trieBuilder.add("-and-", CompoundPart.COMPOUND_PART_AND.getTrieIndex());
+            trieBuilder.add("per-", InitialCompoundPart.INITIAL_COMPOUND_PART_PER.getTrieIndex());
+            trieBuilder.add("square-", PowerPart.POWER_PART_P2.getTrieIndex());
+            trieBuilder.add("cubic-", PowerPart.POWER_PART_P3.getTrieIndex());
+            trieBuilder.add("pow2-", PowerPart.POWER_PART_P2.getTrieIndex());
+            trieBuilder.add("pow3-", PowerPart.POWER_PART_P3.getTrieIndex());
+            trieBuilder.add("pow4-", PowerPart.POWER_PART_P4.getTrieIndex());
+            trieBuilder.add("pow5-", PowerPart.POWER_PART_P5.getTrieIndex());
+            trieBuilder.add("pow6-", PowerPart.POWER_PART_P6.getTrieIndex());
+            trieBuilder.add("pow7-", PowerPart.POWER_PART_P7.getTrieIndex());
+            trieBuilder.add("pow8-", PowerPart.POWER_PART_P8.getTrieIndex());
+            trieBuilder.add("pow9-", PowerPart.POWER_PART_P9.getTrieIndex());
+            trieBuilder.add("pow10-", PowerPart.POWER_PART_P10.getTrieIndex());
+            trieBuilder.add("pow11-", PowerPart.POWER_PART_P11.getTrieIndex());
+            trieBuilder.add("pow12-", PowerPart.POWER_PART_P12.getTrieIndex());
+            trieBuilder.add("pow13-", PowerPart.POWER_PART_P13.getTrieIndex());
+            trieBuilder.add("pow14-", PowerPart.POWER_PART_P14.getTrieIndex());
+            trieBuilder.add("pow15-", PowerPart.POWER_PART_P15.getTrieIndex());
+
+            // Add SI prefixes
+            for (MeasureUnit.SIPrefix siPrefix :
+                    MeasureUnit.SIPrefix.values()) {
+                trieBuilder.add(siPrefix.getIdentifier(), siPrefix.getTrieIndex());
+            }
+
+            // Add simple units
+            for (int i = 0; i < simpleUnits.length; i++) {
+                trieBuilder.add(simpleUnits[i], i + UnitsData.Constants.kSimpleUnitOffset);
+
+            }
+
+            // TODO: Use SLOW or FAST here?
+            trie = trieBuilder.build(StringTrieBuilder.Option.FAST);
+        }
+
+        private static CharsTrie trie = null;
+
+
+        // Set to true when we've seen a "-per-" or a "per-", after which all units
+        // are in the denominator. Until we find an "-and-", at which point the
+        // identifier is invalid pending TODO(CLDR-13700).
+        private boolean fAfterPer = false;
+
+        private String fSource;
+
+        // Tracks parser progress: the offset into fSource.
+        int fIndex = 0;
```

Comment:
Why did you make some of the fields private and others not private (like this one)?

Reply:
I just forgot this one